src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncMethodToClassRewriter.Spilling.vb

```diff
@@ -319,8 +331,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                          fieldAccess.ConstantsInProgressOpt,
                                                          fieldAccess.Type)
 
-                        ' Make sure side effects are checked
-                        builder.AddStatement(Me.F.ExpressionStatement(fieldAccess))
+                        If evaluateSideEffectsHere Then
```

Comment:
>If evaluateSideEffectsHere Then [](start = 24, length = 31)

It looks like C# doesn't evaluate side-effects for an access to a static field. Should we be consistent?

Reply:
I guess ```Unspillable``` check above would catch that.

---
In reply to: [408139568](https://github.com/dotnet/roslyn/pull/42807#discussion_r408139568) [](ancestors = 408139568)